### ISSUE TITLE
❇️ Marta 0.6.1 RC

### ISSUE THREAD

**Issue by yanex** [marta-file-manager/marta-issues#489]
# Marta 0.6.1 RC

I am happy to announce the Marta 0.6.1 release candidate. There are a lot of new features coming out, and I hope you will like it. Please report any bugs you find to this thread (or file the separate issues).

### Marco
Marta doesn't use JSON for the configuration files and themes anymore.  
Meet [Marco](https://github.com/marta-file-manager/MarcoKit/blob/master/docs/MARCO.md) – it perfectly suits Marta needs.

### Integrated preferences editor
'Preferences' action now opens the embedded two-pane Marco editor. 0.6.1 supports only the basic code insight (highlighting, gutter icons). Auto-completion and additional diagnostics will be available in the next versions.

### File coloring
Now you can enable coloring for directories, package bundles, and symbolic links:

```
behavior {
    table {
        colorHighlighting true
    }
}
```

Extensible file coloring will be available in the next versions.

### Action arguments
Now you can pass arguments for the certain actions right from the preferences file. For example, here is how you can bind `⎇H` to open the home directory:

```
keyBindings {
    "Alt+H" {
        id "core.open.directory"
        src "~"
    }
}
```

## Download

Press the button below to download the release candidate:

<a href="https://github.com/marta-file-manager/marta-issues/files/2828438/Marta.dmg.zip"><img alt="Download Marta 0.6.1 RC" src="https://user-images.githubusercontent.com/95996/52214774-85bc4800-28a3-11e9-903a-e842404f93e2.png" width="160" height="60" /></a>

## Changelog
* **New features**
    * New configuration format – [Marco](https://github.com/marta-file-manager/MarcoKit/blob/master/docs/MARCO.md) [#223](https://github.com/marta-file-manager/marta-issues/issues/223)
    * Integrated preferences editor
    * Support action arguments [#458](https://github.com/marta-file-manager/marta-issues/issues/458)
    * Optional color highlighting for files [#108](https://github.com/marta-file-manager/marta-issues/issues/108)
    * Quick Select: highlight the filter on zero results [#478](https://github.com/marta-file-manager/marta-issues/issues/478)
    * Show progress indicator on entering directories on slow external drives [#220](https://github.com/marta-file-manager/marta-issues/issues/220)
    * Show selected item count in the status bar [#354](https://github.com/marta-file-manager/marta-issues/issues/354)
    * Save state of all opened windows [#470](https://github.com/marta-file-manager/marta-issues/issues/470)
    * Support Shift+Home, Shift+PageUp hotkeys [#464](https://github.com/marta-file-manager/marta-issues/issues/464)
    * Use left and right arrows to go all the way up and down [#409](https://github.com/marta-file-manager/marta-issues/issues/409)
    * Rename "Copy" to "Copy to…", add a silent "Copy" command [#387](https://github.com/marta-file-manager/marta-issues/issues/387)
* **Bugfixes**
    * Fix alternative latin layouts (German, Colemak, etc.) [#456](https://github.com/marta-file-manager/marta-issues/issues/456) [#457](https://github.com/marta-file-manager/marta-issues/issues/457)
    * "=" doesn't open the operation queue [#356](https://github.com/marta-file-manager/marta-issues/issues/356)
    * Symlinks become absolute when copying a folder with symlinks [#417](https://github.com/marta-file-manager/marta-issues/issues/417)
    * Most of the actions should not be triggered when the terminal is active [#352](https://github.com/marta-file-manager/marta-issues/issues/352)
    * Favorites are not loaded [#455](https://github.com/marta-file-manager/marta-issues/issues/455)
    * "Open with…" opens only the first selected file [#459](https://github.com/marta-file-manager/marta-issues/issues/459)
    * Inaccurate scrolling in PageUp/PageDown [#465](https://github.com/marta-file-manager/marta-issues/issues/465)
    * Remove "Open as…" [#461](https://github.com/marta-file-manager/marta-issues/issues/461)
    * Reveal in Finder doesn't work for the 'go-up' directory [#462](https://github.com/marta-file-manager/marta-issues/issues/462)
    * Key binding doesn't work with Ctrl+Tab key [#184](https://github.com/marta-file-manager/marta-issues/issues/184)
    * Fix delay between consequent "Toggle hidden files" executions [#306](https://github.com/marta-file-manager/marta-issues/issues/306)
    * Quick Select: recognize an underscore as a word separator [#481](https://github.com/marta-file-manager/marta-issues/issues/481)
    * Double click on the title bar should zoom the window [#476](https://github.com/marta-file-manager/marta-issues/issues/476)
    * Simplify keyboard navigation in the “Target already exists” alert window [#477](https://github.com/marta-file-manager/marta-issues/issues/477)

**Comment by krupenja** [comment]
Quick round-up of what I've noticed so far:

1. Tooltips in the built-in editor are not very intuitive. It took me 10 min to figure out that cogwheel icons are there to show properties descriptions. Maybe change the icon to somethingelse? Circled (i) icon?
2. Old setting to show action bar was not automatically converted into the new format and was in non-working state:

```
    actionBar {   
        show false
    }
```

Had to manually change it to the proper `showActionBar true`

3. Typo in the "Value of wrong type" popup, there is a closing bracket (after `boolean`) and no opening bracket:

![image](https://user-images.githubusercontent.com/26354715/52220990-d1bdbb80-28a8-11e9-918f-b565ad424116.png)

4. Is state.jsion still being used? If not, would be great to mention it in the changelog so users know that they can delete that file :)
5. `colorHighlighting true` does not seem to work. After enabling it, I still only get hidden files highlighted in grey. No highlighting for folders or symlinks.
6. Animated highlighting for #478 is OK but I think the flash speed is too rapid, you can barely notice it. I would suggest making the animation a bit slower. Alternatively, the red highlighting could stay on indefinitely until you enter a string which matches something?

Hope this helps :) Please let me know if you want any of these as separate issues.

**Comment by yanex** [comment]
Thanks for the quick feedback!

1. "Cogwheel" means this is a configuration option. But I aggree it may be not intuitive. I will add an explanatary screen that will show when the preferences editor is opened for the first time.
2, 3. Thanks! Fixed.
4. Marta doesn't use `stat.json` anymore, you are free to delete it. Its contents moved to `Library/Preferences` (aka 'user defaults').
5. Looks like color highlighting doesn't work in any theme besides the default, "Kon". Will be fixed in the release version. Thanks!
6. Thanks for the suggestion! I made animation twice slower. Also, I will experiment with the highlighting in the later versions of Marta (and will try the 'stay indefinitely' mode).
